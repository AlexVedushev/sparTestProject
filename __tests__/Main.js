import 'react-native';
import React from 'react';
import { shallow, mount, render } from 'enzyme';
import 'jsdom-global/register';
import "isomorphic-fetch"
import PropTypes from 'prop-types';
import sinon from 'sinon';
import MainScreen from '../screens/MainSection/Main';
import MainStackNavigator from '../screens/MainSection/MainStackNav';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('Testing Main components:', () => {

  it('renders correctly', () => {
    const tree = renderer.create(
      <MainScreen />
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render again', () => {
      const wrapper = shallow(
          <MainScreen />
      );
          console.log(wrapper.debug());
      expect(wrapper).toMatchSnapshot();
  });


  // it('Test TextInput default state', () => {
  //    const wrapper = shallow(<CartListItem />);
  //    expect((wrapper.find('TextInput')).node.props.value).toEqual('1');
  //  });
  //
  // it('render shown buttons', () => {
  //   const wrapper = shallow(<CartListItem />);
  //   //console.log(wrapper.debug());
  //   expect((wrapper.find('TouchableHighlight')).length).toEqual(2);
  // });
  //
  // it('renders as expected', () => {
  //   const wrapper = shallow(
  //     <CartListItem textValue={1} />
  //   );
  //   //console.log(wrapper.debug());
  //   expect(wrapper).toMatchSnapshot();
  //   wrapper.setProps(renderer: { textValue: 2 });
  //   expect(wrapper).toMatchSnapshot();
  // });

});

describe('Testing MainStackNavigator components:', () => {


  it('renders correctly', () => {
    const tree = renderer.create(
      <MainStackNavigator />
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });



  // it('should render again', () => {
  //     const wrapper = shallow(
  //         <MainStackNavigator />
  //     );
  //         console.log(wrapper.debug());
  //     expect(wrapper).toMatchSnapshot();
  // });
});
