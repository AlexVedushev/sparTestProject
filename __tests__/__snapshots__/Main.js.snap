// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Main components: renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingTop": 20,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;

exports[`Testing Main components: should render again 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
      Object {
        "flex": 1,
        "paddingTop": 20,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="small"
    />
  </View>,
  "nodes": Array [
    <View
      style={
        Object {
          "flex": 1,
          "paddingTop": 20,
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#999999"
        hidesWhenStopped={true}
        size="small"
      />
    </View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <MainScreen />,
    "_instance": MainScreen {
      "_onPressCellWithId": [Function],
      "context": Object {},
      "props": Object {},
      "refs": Object {},
      "state": Object {
        "dateList": Array [],
        "isLoadingProducts": true,
        "isLoadingSections": true,
      },
      "updater": Updater {
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <View
      style={
        Object {
          "flex": 1,
          "paddingTop": 20,
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#999999"
        hidesWhenStopped={true}
        size="small"
      />
    </View>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <MainScreen />,
}
`;
